File: backend/internal/domain/audit.go
package domain

import "github.com/google/uuid"

// ActivityLogEntry represents a single audit event to be recorded.
type ActivityLogEntry struct {
	WorkspaceID uuid.UUID
	UserID      *uuid.UUID
	Action      string         // e.g. "content.created"
	EntityType  string         // e.g. "content"
	EntityID    *uuid.UUID
	Metadata    map[string]any
}

File: frontend/packages/types/src/audit.ts
import type { UUID } from './common';

export interface ActivityLogEntry {
  workspace_id: UUID;
  user_id?: UUID | null;
  action: string; // e.g. "content.created"
  entity_type: string; // e.g. "content"
  entity_id?: UUID | null;
  metadata?: Record<string, unknown> | null;
}
